src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Linq;
+using NBitcoin;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+
+namespace Stratis.Bitcoin.P2P
+{
+    /// <summary>
+    /// The connector used to connect to peers added via peer discovery.
+    /// </summary>
+    public sealed class PeerConnectorDiscovery : PeerConnector
+    {
+        /// <summary>Constructor used for unit testing.</summary>
+        public PeerConnectorDiscovery(NodeSettings nodeSettings, IPeerAddressManager peerAddressManager)
+            : base(nodeSettings, peerAddressManager)
+        {
+        }
+
+        /// <summary>Constructor used by <see cref="Connection.ConnectionManager"/>.</summary>
+        public PeerConnectorDiscovery(PeerConnectorContext context) : base(context)
+        {
+            this.GroupSelector = WellKnownPeerConnectorSelectors.ByNetwork;
+            this.MaximumNodeConnections = 8;
+            this.Requirements = new NetworkPeerRequirement
+            {
+                MinVersion = context.NodeSettings.ProtocolVersion,
+                RequiredServices = NetworkPeerServices.Network
+            };
+        }
+
+        /// <inheritdoc/>
+        internal override NetworkAddress FindPeerToConnectTo()
+        {
+            int peerSelectionFailed = 0;
+
+            NetworkAddress peer = null;
+
+            while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested && peerSelectionFailed < 50)
+            {
+                peer = this.peerAddressManager.SelectPeerToConnectTo();
+
+                if (!peer.Endpoint.Address.IsValid())
+                {
+                    peerSelectionFailed++;
+                    continue;
+                }
+
+                // If the peer exists in the -addnode collection don't 
```

Comment:
AHHHH very good way to overcome this issue good job mate

Reply:
🥇